dali/operators/audio/nonsilence_op_gpu.cu

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/common/find/find_first_last_gpu.cuh"
+#include "dali/kernels/reduce/reduce_gpu.h"
+#include "dali/kernels/signal/decibel/decibel_calculator.h"
+#include "dali/kernels/signal/moving_mean_square_gpu.h"
+#include "dali/operators/audio/nonsilence_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+namespace {
+
+struct threshold_cutoff_db {
+  float thresh_;
+
+  DALI_HOST_DEV DALI_FORCEINLINE threshold_cutoff_db() = default;
+  DALI_HOST_DEV DALI_FORCEINLINE explicit threshold_cutoff_db(float cutoff_db, float ref) {
+    kernels::signal::DecibelToMagnitude<float> db2mag(10.f, ref);
+    thresh_ = db2mag(cutoff_db);
+  }
+
+  DALI_HOST_DEV DALI_FORCEINLINE bool operator()(float x) const noexcept {
+    return x >= thresh_;
+  }
+};
+
+template <typename Idx>
+struct nonsilent_region_fmt {
```

Comment:
I've read the entire class and still don't know what `fmt` stands for (usually it's "format", but it doesn't seem like it here).